src/app/content/utils.ts

```diff
@@ -59,3 +61,77 @@ export const scrollTocSectionIntoView = (sidebar: HTMLElement | undefined, activ
 
   sidebar.scrollTop = scrollTarget.offsetTop;
 };
+
+export const findArchiveTreeSection = (
+  book: {tree: ArchiveTree},
+  pageId: string
+): LinkedArchiveTreeSection | undefined =>
+  flattenArchiveTree(book.tree).find((section) =>
+    stripIdVersion(section.shortId) === stripIdVersion(pageId)
+    || stripIdVersion(section.id) === stripIdVersion(pageId)
+  );
+
+const getUrlParamForPageTitle = (section: LinkedArchiveTreeSection): string => {
+  const splitTitleParts = (str: string) => {
+    const match = str
+      // remove html tags from tree title
+      .replace(/<[^>]+>/g, '')
+      // split out section number from title
```

Comment:
I wonder if having `.` in the URL will cause people problems... It _might_ look like the file has a _really_ long file extension

Reply:
ok, I don't think it's a big issue... periods are probably fine. I assume/hope that LMS' and things handle periods in URLs (like autodetecting what is a URL from a big paragraph of text). That might be something we can test before release? Thoughts @kjdav ?